app/models/annotation_category.rb

```diff
@@ -27,4 +31,21 @@ def self.add_by_row(row, assignment, current_user)
       end
     end
   end
+
+  def update_annotation_text_deductions
+    return unless flexible_criterion_id_changed?
+    prev_criterion = previous_changes["flexible_criterion_id"].first
+    new_criterion = previous_changes["flexible_criterion_id"].second
+    if prev_criterion != new_criterion
+      if new_criterion.nil?
+        self.annotation_texts.each do |text|
+          text.update!(deduction: nil)
+        end
+      else
+        self.annotation_texts.each do |text|
+          text.update!(deduction: 0)
```

Comment:
This is interesting, let's calculate this by scaling the deduction according to the ratio of the previous criterion's `max_mark` and the new `max_mark` (unless the previous criterion was `nil`).